flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/restart/NoOrFixedIfCheckpointingEnabledRestartStrategyFactory.java

```diff
@@ -18,12 +18,14 @@
 
 package org.apache.flink.runtime.executiongraph.restart;
 
+import java.time.Duration;
+
 /**
  * Default restart strategy that resolves either to {@link NoRestartStrategy} or {@link FixedDelayRestartStrategy}
  * depending if checkpointing was enabled.
  */
 public class NoOrFixedIfCheckpointingEnabledRestartStrategyFactory extends RestartStrategyFactory {
-	private static final long DEFAULT_RESTART_DELAY = 0;
+	private static final long DEFAULT_RESTART_DELAY = Duration.ofSeconds(1L).toMillis();
```

Comment:
can we make this dependent on the default delay for the FixedDelayStrategy?

Reply:
We would need to parse the string into a long. I would like to avoid to do this in a static block because if this fails, then the JVM process will fail to start.

Reply:
But we could parse it when creating the `NoOrFixedDelayRestartStrategy` in the `create` method.

Reply:
Did you ultimately decide against it? @tillrohrmann

Reply:
Yes, we can change it once we have replaced Scala `Duration` parsing logic. I didn't want to further spread the usage of the Scala classes in the `flink-runtime` module.